src/main/java/unrc/dose/Proposition.java

```diff
@@ -1,18 +1,417 @@
+     */
+    public void setDistance(final int distance) {
+        this.set("distance", distance);
+    }
+
+    /**
+     * Set cantTestPassed value.
+     * @param testPassed reference the new amount of tests passed
+     */
+    public void setCantTestPassed(final int testPassed) {
+        this.set("cantTestPassed", testPassed);
+    }
+
+    /**
+     * Method for creates new proposition in the database.
+     * @param userId reference the user for the new proposition.
+     * @param challengeId reference the challenge for the new proposition.
+     * @return the proposition created in the database.
+     */
+    public static Proposition newProposition(
+        final int userId,
+        final int challengeId) {
+
+        Challenge currentChallenge = Challenge.findById(challengeId);
+        User currentUser = User.findById(userId);
+
+        if (currentChallenge == null || currentUser == null) {
+            throw new IllegalArgumentException();
+        }
+
+        Proposition newProposition = new Proposition();
+        newProposition.set("user_id", currentUser.getInteger("id"));
+        newProposition.set("challenge_id", currentChallenge.getInteger("id"));
+        newProposition.set("source", currentChallenge.get("source"));
+        newProposition.set("isSolution", false);
+        newProposition.set("distance", 0);
+        newProposition.set("cantTestPassed", 0);
+        newProposition.saveIt();
+        return newProposition;
+    }
+
+    /**
+     * Get the solutions for a challenge by a specific user..
+     * @param userId reference the id of a user
+     * @param challengeId reference the id of a challenge
+     * @return List of proposition for user in challenge
+     */
+    public static LazyList<Proposition> getChallengeSolutionsByUser(
+            final Integer userId, final Integer challengeId) {
+        return Proposition.where("user_id = ? and challenge_id = ? "
+                + "and isSolution = ?", userId, challengeId, true);
+    }
+
+    /**
+     * Save a user's solution in a challenge.
+     * @param proposedCode reference the new code of the proposed solution
+     * @param distanceObtained reference the new distance obtained
+     */
+    public void saveSolution(
+        final String proposedCode,
+        final int distanceObtained) {
+        this.setSource(proposedCode);
+        this.set("distance", distanceObtained);
+        this.set("isSolution", true);
+    }
+
+    /**
+     * save proposed solution in a challenge.
+     * @param sourceCurrent reference the new code of the proposed solution
+     */
+    public void saveProposedSolution(final String sourceCurrent) {
+        this.set("source", sourceCurrent);
+    }
+
+    /**
+     * This method find the proposition that is not a solution yet.
+     * @param userId represents the user
+     * @param challengeId represent the challenge
+     * @return null if there is no proposition or the proposition,
+     *  or the proposition found.
```

Comment:
This method should never return `null`, if no object is found then the result should be an empty list